### PR TITLE
Customer.io webhook

### PR DESCRIPTION
**Pull request by mohan-h**
Initial commit

### CODE REVIEW

File: help/sources/tutorials/ui/create/marketing-automation/customerio-webhook.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+## Getting started {#getting-started}
+
+This tutorial requires a working understanding of the following components of Experience Platform:
+
+* [[!DNL Experience Data Model (XDM)] System](../../../../../xdm/home.md): The standardized framework by which [!DNL Experience Platform] organizes customer experience data.
+  * [Basics of schema composition](../../../../../xdm/schema/composition.md): Learn about the basic building blocks of XDM schemas, including key principles and best practices in schema composition.
+  * [Schema Editor tutorial](../../../../../xdm/tutorials/create-schema-ui.md): Learn how to create custom schemas using the Schema Editor UI.
+* [[!DNL Real-Time Customer Profile]](../../../../../profile/home.md): Provides a unified, real-time consumer profile based on aggregated data from multiple sources.
+
+## Prerequisites - Sample JSON to define source schema for [!DNL Customer.io] {#prerequisites-json-schema}
```
Comment (carlojavier) [inline help/sources/tutorials/ui/create/marketing-automation/customerio-webhook.md]: Let's add a short intro here and move the JSON part into its own H3. So something like:
## Prerequisites 
The following section provides information on prerequisites to complete before you can create a [!DNL Customer.io]  source connection.

### Sample JSON to define the source schema for [!DNL Customer.io]

File: help/sources/tutorials/ui/create/marketing-automation/customerio-webhook.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+      "email": "test@example.com",
+      "cio_id": "d9c106000001"
+    }
+  }
+}
+```
+
+### Create a Platform schema for [!DNL Customer.io]
+
+Before creating a [!DNL Customer.io] source connection, you must also ensure that you first create a Platform schema to use for your source. See the tutorial on [creating a Platform schema](../../../../../xdm/schema/composition.md) for comprehensive steps on how to create a schema.
```
Comment (carlojavier) [inline help/sources/tutorials/ui/create/marketing-automation/customerio-webhook.md]: We can remove the "Before creating a [!DNL Customer.io] source connection," here and start with "You must also ensure..."

File: help/sources/tutorials/ui/create/marketing-automation/customerio-webhook.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+![Platform UI screenshot showing an example schema for Customer.io](../../../../images/tutorials/create/marketing-automation/customerio-webhook/schema.png)
+
+### Integrate [!DNL Customer.io] with your webhook
+
```
Comment (carlojavier) [inline help/sources/tutorials/ui/create/marketing-automation/customerio-webhook.md]: We can remove this section, just an optional space for where webhook integration can be added to

File: help/sources/tutorials/ui/create/marketing-automation/customerio-webhook.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+| Target Field | Description |
+| --- | --- |
+| `object_type` | Object Type, refer to the [!DNL Customer.io] [events](https://customer.io/docs/webhooks/#events) documentation for the supported types. |
+| `id` | Object Id. |
+| `email` | Email address associated with the object. |
+| `event_id` | Unique identifier of the event. |
+| `cio_id` | [!DNL Customer.io] identifier for the event.|
+| `metric` | Event Type, refer to the [!DNL Customer.io] [events](https://customer.io/docs/webhooks/#events) documentation for the supported types. |
+| `timestamp` | Timestamp when the event occurred. |
```
Comment (carlojavier) [inline help/sources/tutorials/ui/create/marketing-automation/customerio-webhook.md]: Good call on the table here! One small suggestion: can we make the descriptions into full sentences? Just add a "The" before each of them.

File: help/sources/tutorials/ui/create/marketing-automation/customerio-webhook.md [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+>[!NOTE]
```
Comment (carlojavier) [inline help/sources/tutorials/ui/create/marketing-automation/customerio-webhook.md]: Let's turn this into an important tag:
>[!IMPORTANT]
>
>Do not map...

File: help/sources/tutorials/ui/create/marketing-automation/customerio-webhook.md [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+
+![The review step of the sources workflow.](../../../../images/tutorials/create/marketing-automation/customerio-webhook/review.png)
+
+## Get your streaming endpoint URL {#get-streaming-endpoint-url}
+
+With your streaming dataflow created, you can now retrieve your streaming endpoint URL. This endpoint will be used to subscribe to your webhook, allowing your streaming source to communicate with Experience Platform. 
+
+Next, you will require to construct a URL using the API details generated within Platform which will be used to configure the webhook on [!DNL Customer.io].
+
+Note down the values of the fields below:
```
Comment (carlojavier) [inline help/sources/tutorials/ui/create/marketing-automation/customerio-webhook.md]: Let's rephrase this to: "You must retrieve the following in order to construct a URL:"

File: help/sources/tutorials/ui/create/marketing-automation/customerio-webhook.md [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+* **[!UICONTROL Streaming endpoint]**
+
+To retrieve your **[!UICONTROL Dataflow ID]** and **[!UICONTROL Streaming endpoint]**, go to the [!UICONTROL Dataflow activity] page of the dataflow that you just created and copy the details from the bottom of the [!UICONTROL Properties] panel.
+
+![The streaming endpoint in dataflow activity.](../../../../images/tutorials/create/marketing-automation/customerio-webhook/endpoint-test.png)
+
+Construct a URL using these values following the pattern below:
+```{STREAMING_ENDPOINT}?x-adobe-flow-id={DATAFLOW_ID}```
+
+Example of a constructed webhook URL: ``https://dcs.adobedc.net/collection/febc116d22ba0ea2868e9c93b199375302afb8a589617700991bb8f3f0341ad7?x-adobe-flow-id=439b3fc4-3042-4a3a-b5e0-a494898d3fb0``
```
Comment (carlojavier) [inline help/sources/tutorials/ui/create/marketing-automation/customerio-webhook.md]: Once you have retrieved your streaming endpoint and dataflow ID, build a URL based on the following pattern: `{STREAMING_ENDPOINT}?x-adobe-flow-id={DATAFLOW_ID}`. For example, a constructed webhook URL may look like: `https://dcs.adobedc.net/collection/febc116d22ba0ea2868e9c93b199375302afb8a589617700991bb8f3f0341ad7?x-adobe-flow-id=439b3fc4-3042-4a3a-b5e0-a494898d3fb0`

File: help/sources/tutorials/ui/create/marketing-automation/customerio-webhook.md [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+
+Refer to the [!DNL Customer.io] [events](https://customer.io/docs/webhooks/#events) documentation section to understand in detail on the various events that can be selected, each events' message will be ingested into Platform when any [!DNL Customer.io] action event trigger criteria are satisfied.
+
+## Additional resources
+
+The sections below provide additional resources that you can refer to when using the [DNL Customer.io] source.
+
+### Guardrails {#guardrails}
+
+Refer to the [DNL Customer.io] [Timeouts and failures](https://customer.io/docs/webhooks/#timeouts-and-failures) page.
```
Comment (carlojavier) [inline help/sources/tutorials/ui/create/marketing-automation/customerio-webhook.md]: Let's revise this to: "For information on guardrails, please refer to the [[!DNL Customer.io] timeouts and failures page](https://customer.io/docs/webhooks/#timeouts-and-failures).

File: help/sources/tutorials/ui/create/marketing-automation/customerio-webhook.md [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+
+Example of a constructed webhook URL: ``https://dcs.adobedc.net/collection/febc116d22ba0ea2868e9c93b199375302afb8a589617700991bb8f3f0341ad7?x-adobe-flow-id=439b3fc4-3042-4a3a-b5e0-a494898d3fb0``
+
+## Next steps {#next-steps}
+
+By following this tutorial you have successfully [configured a dataflow to bring data into Platform](https://experienceleague.adobe.com/docs/experience-platform/sources/ui-tutorials/dataflow/crm.html).
+
+Next, login to your account on [[!DNL Customer.io]](https://customer.io/) and create a reporting webhook by following this [tutorial](https://customer.io/docs/webhooks/#setup).
+
+Use the [URL](#get-streaming-endpoint-url) that was constructed in the [!UICONTROL WEBHOOK ENDPOINT] field.
```
Comment (carlojavier) [inline help/sources/tutorials/ui/create/marketing-automation/customerio-webhook.md]: Let's move the parts about webhooks out of Next steps and into its own heading between "get streaming endpoint URL" and "Next steps", maybe we can call it "Integrate Customer.io Webhooks" or something similar and update the link back in the connector overview page

Because this one is not Platform UI, let's replace "[!UICONTROL WEBHOOK ENDPOINT]" with [!DNL WEBHOOK ENDPOINT]

File: help/sources/tutorials/api/create/marketing-automation/customerio-webhook.md [hunk 10]
```diff
@@ -0,0 +1,448 @@
...
+          "format": "json"
+      }
+    }'
+```
+
+| Property | Description |
+| --- | --- |
+| `name` | The name of your source connection. Ensure that the name of your source connection is descriptive as you can use this to look up information on your source connection. |
+| `description` | An optional value that you can include to provide more information on your source connection. |
+| `baseConnectionId` | The base connection ID of [!DNL Customer.io]. This ID was generated in an earlier step. |
```
Comment (carlojavier) [inline help/sources/tutorials/api/create/marketing-automation/customerio-webhook.md]: We can remove this line

File: help/sources/tutorials/api/create/marketing-automation/customerio-webhook.md [hunk 11]
```diff
@@ -0,0 +1,448 @@
...
+
+The last step towards bringing data from [!DNL Customer.io] to Platform is to create a dataflow. By now, you have the following required values prepared:
+
+* [Source connection ID](#source-connection)
+* [Target connection ID](#target-connection)
+* [Mapping ID](#mapping)
+
+A dataflow is responsible for scheduling and collecting data from a source. You can create a dataflow by performing a POST request while providing the previously mentioned values within the payload.
+
+To schedule an ingestion, you must first set the start time value to epoch time in seconds. Then, you must set the frequency value to one of the five options: `once`, `minute`, `hour`, `day`, or `week`. The interval value designates the period between two consecutive ingestions however, creating a one-time ingestion does not require an interval to be set. For all other frequencies, the interval value must be set to equal or greater than `15`.
```
Comment (carlojavier) [inline help/sources/tutorials/api/create/marketing-automation/customerio-webhook.md]: Let's remove this part. Looks like we won't need it for streaming. I will also remove it from the template.

File: help/sources/connectors/marketing-automation/customerio-webhook.md [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+* Slack Events
+* Webhook Events
+
+Refer to the [[!DNL Customer.io] Reporting Webhook events](https://customer.io/docs/webhooks/#events) page for a list of  events that are available via webhook.
+
+## Prerequisites {#prerequisites}
+
+Before you can create a [!DNL Customer.io] source connection, you must first ensure that you have the following:
+
+A validated [!DNL Customer.io] account. If you do not have one already see the [[!DNL Customer.io] signup page](https://fly.customer.io/signup) to register and create your account.
```
Comment (carlojavier) [inline help/sources/connectors/marketing-automation/customerio-webhook.md]: (Pls let me know if the meaning here gets altered entirely) Let's remove "validated" here because it can be confusing when read with the following sentence, that also talks about getting your account validated. If it's fine to remove, let's go with:
"A [!DNL Customer.io] account. If you do not have one already, read the [[!DNL Customer.io] signup page](https://fly.customer.io/signup) to register and create your account.

File: help/sources/connectors/marketing-automation/customerio-webhook.md [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+
+Once you have successfully registered, you will also need to have your account validated. Follow the steps documented on the [[!DNL Customer.io] Account verification](https://customer.io/docs/account-verification/) page to complete the process.
+
+Later, after you set up the [!DNL Customer.io] source, you will set up a Reporting Webhook to inform Platform about [!DNL Customer.io] events as explained in the [Get your streaming endpoint URL](../../tutorials/ui/create/marketing-automation/customerio-webhook.md#set-up-webhook) and the [Next Steps](../../tutorials/ui/create/marketing-automation/customerio-webhook.md#next-steps) sections. Webhooks can notify you immediately when customer attributes change or when people open your messages, and send this information to your [!DNL Customer.io] source.
```
Comment (carlojavier) [inline help/sources/connectors/marketing-automation/customerio-webhook.md]: This feels like a big step so let's have it as an H3:

### Set up Reporting Webhook

Once you have successfully created your dataflow, you must set up a Reporting Webhook to inform Platform about [!DNL Customer.io] events. Webhooks can notify you immediately when customer attributes change or when people open your messages, and send this information to your [!DNL Customer.io] source. For more information, read the tutorials on [getting your streaming endpoint URL](../../tutorials/ui/create/marketing-automation/customerio-webhook.md#set-up-webhook) and [setting up a Customer.io Webhook](../../tutorials/ui/create/marketing-automation/customerio-webhook.md#next-steps).

**Review by carlojavier — CHANGES_REQUESTED** [review]
looking good! thanks for making the source overview page too! pls see comments

File: help/sources/connectors/marketing-automation/customerio-webhook.md (hunk 13 above)
Comment (mohan-h) [reply help/sources/connectors/marketing-automation/customerio-webhook.md]: Thanks

**Comment by carlojavier** [comment]
@mohan-h thanks for the revisions. This PR has been copied over to our internal repo
